kong/tracing/instrumentation.lua

```diff
@@ -72,33 +72,32 @@ function _M.balancer(ctx)
     local try = balancer_tries[i]
     span = tracer.start_span("balancer try #" .. i, {
       kind = 3, -- client
-      start_time_ns = try.balancer_start * 100000,
+      start_time_ns = try.balancer_start * 1e6,
       attributes = {
-        ["kong.balancer.state"] = try.state,
-        ["http.status_code"] = try.code,
         ["net.peer.ip"] = try.ip,
         ["net.peer.port"] = try.port,
       }
     })
 
-    if i < try_count then
+    if try.state then
+      span:set_attribute("http.status_code", try.code)
       span:set_status(2)
     end
 
     -- last try
-    if i == try_count then
-      local upstream_finish_time = (ctx.KONG_BODY_FILTER_ENDED_AT or 0) * 100000
+    if i == try_count and try.state == nil then
+      local upstream_finish_time = ctx.KONG_BODY_FILTER_ENDED_AT and ctx.KONG_BODY_FILTER_ENDED_AT * 1e6
       span:finish(upstream_finish_time)
 
     else
       if not upstream_connect_time then
-        upstream_connect_time = split(var.upstream_connect_time, ",", "jo")
+        upstream_connect_time = split(var.upstream_connect_time, ", ", "jo")
       end
```

Comment:
It seems a rare condition where this lazy evaluation would work: almost every time the `else` branch is entered.
Could we just evaluate this before the loop?

Reply:
Also, is the format of this variable fixed? Or maybe we should do this:
```lua 
split(var.upstream_connect_time, ",\\s*", "jo")
```

Reply:
https://github.com/nginx/nginx/blob/bfc5b35827903a3c543b58e4562db8b62021c164/src/http/ngx_http_upstream.c#L5758-L5760
According to the Nginx implementation, it will concat the string with `, `.

```lua
local ngx_re = require "ngx.re"
local split = ngx_re.split

local test_str = "1000, 1000, 1000"

for _, v in ipairs(split(test_str, ", ", "jo")) do
    print(v)
end
```
output: 
```bash
$ resty test/split.lua 
1000
1000
1000
```